### DISCUSSION TITLE
How do I require a review of Pull Requests (and allow admin override)?

### DISCUSSION THREAD

**Discussion by DavidThielen** [community/community#136200] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

Hi all;

I had this working (implemented a year ago after hours of reading the docs and experimenting). And then I moved my repository to an organization and didn't copy my rule settings first (yes - stupid). I'm hoping someone here can tell me how to set this up again so I don't need to once again invest a couple of hours on this. (For background I'm a developer, not a DevOps person.)

What I want is:
1. For all branches except master - do anything.
2. The only way to update master is to branch, edit, check it in, create a Pull Request.
3. A Pull Request must be approved by one other person to be merged.
4. An admin (that's me) can override the requirement for another to first approve a PR.

thanks - dave

**Comment by SjxSubham** [comment]
Navigate to the Repository Settings:
   - Go to your repository on GitHub.
   - Click on `Settings` in the repository menu.

Access Branch Protection Rules:
   - In the left sidebar, click on `Branches`.
   - Under the "Branch protection rules" section, click on `Add rule`.

Set Up the Rule for the `master` Branch:
   - In the "Branch name pattern" field, type `master`.
   - Check the box for `Require pull request reviews before merging`.
     - Set the number of required approving reviews to `1`.
   - Check the box for `Include administrators` to ensure that the rules apply to admins as well.
   - Optionally, you can check `Allow administrators to bypass branch protection rules` to allow admin override.


Save Changes:
   - Click on `Create` or `Save changes`.


 Summary of Settings

- Branch name pattern: `master`
- Require pull request reviews before merging: Checked
  - Number of required approving reviews:  `1`
- Include administrators:  Checked
- Allow administrators to bypass branch protection rules:  Optionally Checked

Following these steps will enforce that all changes to the `master` branch must go through a pull request, be reviewed by at least one other person, and allow you as an admin to override the rules if necessary.

I hope this helps :)

**Comment by DavidThielen** [comment, marked as answer]
@SjxSubham It didn't have it exactly as you specified above. Is the following correct?

thanks - dave

![Ruleset_1](https://github.com/user-attachments/assets/9c110381-8b81-43fe-b436-7967946f722a)

![Ruleset_2](https://github.com/user-attachments/assets/67b14f0e-c27e-4884-85d5-e0318343d0ac)

**Comment by SjxSubham** [comment]
Yes, all good U have already checked all of them correctly...

**Comment by SjxSubham** [comment]
have u got ur solution??
then Marked the solution As answer ☺️

**Reply by DavidThielen** [reply]
@SjxSubham It worked - thank you

**Reply by SjxSubham** [reply]
Hi @DavidThielen if it's worked then mark my comment/solution as answered (U hv marked ur ans as answered)... 
:-)

It will be my pleasure...
Thank U...

**Reply by DavidThielen** [reply]
@SjxSubham It is marked as the answer already. Is there something else I should do?

thanks - dave

**Reply by SjxSubham** [reply]
hello @DavidThielen , U have marked ur reply as answer.
I'm saying that omit that one ... and select my Solution as answered 
then U will see the message that "_Answered by SjxSubham_" which is currently  showing"_Answered by DavidThielen "_

**Reply by DavidThielen** [reply]
@SjxSubham Happy to do so, but I can't find a way to do it. It looks like once an answer is selected, that's it. Sorry.
